lib/archethic/p2p/message/start_mining.ex

```diff
@@ -4,75 +4,129 @@ defmodule Archethic.P2P.Message.StartMining do
 
   This message is initiated by the welcome node after the validation nodes election
   """
-  @enforce_keys [:transaction, :welcome_node_public_key, :validation_node_public_keys]
-  defstruct [:transaction, :welcome_node_public_key, :validation_node_public_keys]
+  @enforce_keys [
+    :transaction,
+    :welcome_node_public_key,
+    :validation_node_public_keys,
+    :network_chains_view_hash,
+    :p2p_view_hash
+  ]
+  defstruct [
+    :transaction,
+    :welcome_node_public_key,
+    :validation_node_public_keys,
+    :network_chains_view_hash,
+    :p2p_view_hash
+  ]
 
   alias Archethic.Crypto
   alias Archethic.Mining
   alias Archethic.Utils
   alias Archethic.TransactionChain.Transaction
   alias Archethic.P2P.Message.Ok
   alias Archethic.P2P.Message.Error
+  alias Archethic.SelfRepair.NetworkChain
+  alias Archethic.SelfRepair.NetworkView
 
   require Logger
 
   @type t :: %__MODULE__{
           transaction: Transaction.t(),
           welcome_node_public_key: Crypto.key(),
-          validation_node_public_keys: list(Crypto.key())
+          validation_node_public_keys: list(Crypto.key()),
+          network_chains_view_hash: binary(),
+          p2p_view_hash: binary()
         }
 
   @spec process(__MODULE__.t(), Crypto.key()) :: Ok.t() | Error.t()
   def process(
         %__MODULE__{
           transaction: tx = %Transaction{},
           welcome_node_public_key: welcome_node_public_key,
-          validation_node_public_keys: validation_nodes
+          validation_node_public_keys: validation_nodes,
+          network_chains_view_hash: network_chains_view_hash,
+          p2p_view_hash: p2p_view_hash
         },
         _
       ) do
-    with {:election, true} <- {:election, Mining.valid_election?(tx, validation_nodes)},
-         {:elected, true} <-
-           {:elected, Enum.any?(validation_nodes, &(&1 == Crypto.last_node_public_key()))},
-         {:mining, false} <- {:mining, Mining.processing?(tx.address)} do
-      {:ok, _} = Mining.start(tx, welcome_node_public_key, validation_nodes)
-      %Ok{}
-    else
-      {:election, false} ->
-        Logger.error("Invalid validation node election",
-          transaction_address: Base.encode16(tx.address),
-          transaction_type: tx.type
-        )
+    case check_synchronization(network_chains_view_hash, p2p_view_hash) do
```

Comment:
I'm wondering if we can reduce this piece of code.
There is a lot which happening into a single function with multiple calls and with statements.
If we can, it should be great to refactor a bit